woocommerce/google-listings-and-ads
# [WC 8.1] - Fix Selector and Target for Add To Cart

### Changes proposed in this Pull Request:



Closes #2095 

This PR fixes 2 bugs.

- "Added" doesn't match anymore as a valid selector when the product is added to cart
- Change `target` to `currentTarget` Avoiding the click event being triggered in children elements (span) that doesn't contain the data-product-id attribute, and hence, GLA ID is set as undefined. 


See https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget

### Screenshots
https://github.com/woocommerce/google-listings-and-ads/actions/runs/6186698347/job/16794930005


### Detailed test instructions:


1. Run `npm run wp-env:up` (be sure you're using WC 8.1, you can use `npm run -- wp-env run tests-cli -- wp core update --version=8.1.0`) 
2. Run `npm run test:e2e  gtag-events/gtag-events.test.js`
3. Verify that test are passing
4. Install WC 8.1 in your site and also TT2.
5. Go to Shop
6. Click on several products on the text inside the button (so, clicking in the span element inside the button) And Verify gla ID is loaded correctly 


### Additional details:


### Changelog entry

> Fix - Tweak E2E tests for WC 8.1
> Fix - Adjust target on click events preventing GLA ID to be Undefined
